src/common/TrackedOp.h

```diff
@@ -233,28 +233,124 @@ class TrackedOp : public boost::intrusive::list_base_hook<> {
   utime_t initiated_at;
 
   struct Event {
-    utime_t stamp;
-    std::string str;
+    utime_t stamp_;
+    string str_;
+    const char *cstr_ = nullptr;
```

Comment:
This probably won't be the first time we want to optimize for string literals in a container. How about a class which abstracts this. Something like:

````
struct litorstring {
  std::string_view sv;
  litorstring(std::string str) : str(std::move(str)), sv(str) {}
  explicit litorstring(const char *literal) : sv(literal) {}
  litorstring() = delete;
private:
  std::string str;
};
````

Of course there is no type check for literal statically allocated strings unfortunately.

Reply:
probably `std::variant<>` is a better solution, as a tagged union we know which element is valid. unless we provide `operator<()` for `litorstring`.

Reply:
Maybe I'm just old and grumpy, but this all feels like a lot of complexity to avoid having const char* be part of the interface.  I'd much prefer that the interface be explicit about what it expects and the ramifications of using it.

From a higher level: the whole idea of using strings for dynamic Events and mixing them with static Events seems flawed to me here.  I'd much rather we only have static Events with an optional string_view for notes or additional dynamic information that does in fact store a copy of the string with the expectation that:

a) the additional notes/information is unique to that event
b) we know that adding this information to the event is slow, so it should only be used rarely and when necessary.

Reply:
I'm supportive of making the name of the event refer to a static string literal.